### PR TITLE
feat: set autohide to false by default and depreciate the prop on toasts

### PR DESCRIPTION
**Pull request by caitlinmwillington**
## Why

Toast notifications that disappear automatically violates WCAG ['2.2 Enough Time'](https://www.w3.org/WAI/WCAG21/Understanding/enough-time) criteria

## What

This change sets the autohide to false, sets the prop as optional, and deprecates the prop.

### CODE REVIEW

File: packages/notification/src/types.ts [hunk 1]
```diff
@@ -7,11 +7,15 @@ export type ToastNotification = {
...
   title: string
   message: React.ReactNode
-  autohide: boolean
   autohideDelay?: "short" | "long"
   onHide?: () => void
   automationId?: string
   persistent?: boolean
+  /**
+   * **Deprecated:**
+   * @deprecated
```
Comment (dougmacknz) [inline packages/notification/src/types.ts:16]: Probably want to put some info in here

```suggestion
   * @deprecated Automatic hiding is an accessibility anti-pattern, violating WCAG '2.2 Enough Time' criteria. This prop will be removed in an upcoming breaking change.
```

File: packages/notification/src/types.ts [hunk 2]
```diff
@@ -7,11 +7,15 @@ export type ToastNotification = {
   type: NotificationType
   title: string
   message: React.ReactNode
-  autohide: boolean
   autohideDelay?: "short" | "long"
```
Comment (dougmacknz) [inline packages/notification/src/types.ts:10]: We should list this prop as deprecated too

File: packages/notification/src/ToastNotification.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ type ToastNotificationProps = Omit<
 const ToastNotification = ({
   id,
   hideCloseIcon = false,
-  autohide = true,
+  autohide = false,
   autohideDelay = "short",
```
Comment (dougmacknz) [inline packages/notification/src/ToastNotification.tsx:22]: Might as well change the default on this while we're here too

```suggestion
  autohideDelay = "long",
```

File: packages/notification/src/types.ts (hunk 2 above)
Comment (caitlinmwillington) [reply packages/notification/src/types.ts]: Good idea, I went with the same deprecated message as the autohide as it seemed to apply for both but let me know if it should be different!

**Comment by gyfchong** [comment]
You'll need to run `yarn changeset` 😄

**Comment by caitlinmwillington** [comment]
> You'll need to run `yarn changeset` 😄

@gyfchong I tried to run it but was getting error Command "changeset" not found. 🤔 Anything I could be missing?

**Comment by dougmacknz** [comment]
Might just need to `yarn install` @caitlinmwillington ?

**Comment by caitlinmwillington** [comment]
> Might just need to `yarn install` @caitlinmwillington ?
Screenshot 2023-05-05 at 5 26 57 pm

@dougmacknz Yeah I did that :/ 
running `npx changeset` gives me:  `npm ERR! could not determine executable to run`

**Comment by gyfchong** [comment]
Attempt `yarn commit`

**Review by dougmacknz — APPROVED** [review]

**Pull request merged**